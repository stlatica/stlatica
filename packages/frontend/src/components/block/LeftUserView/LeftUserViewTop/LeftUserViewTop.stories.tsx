import { LeftUserViewTop } from "./LeftUserViewTop";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "block/LeftUserViewTop",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return (
    <div className="w-[300px]">
      <LeftUserViewTop />
    </div>
  );
};
