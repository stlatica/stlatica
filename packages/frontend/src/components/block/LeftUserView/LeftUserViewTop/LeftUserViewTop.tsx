import type React from "react";

import { BackButton } from "@/components/button/BackButton";
import { FollowButton } from "@/components/button/FollowButton";

// type LeftUserViewTopProps = {
//   // children: React.ReactNode;
// };

/**
 *
 */
export const LeftUserViewTop: React.FC = () => {
  return (
    <div className="flex flex-col">
      <div className="grid h-[170px] grid-cols-10 ">
        <div className="col-span-2 ">
          <div className="h-full">
            <a href="/">
              <BackButton />
            </a>
          </div>
        </div>
        <div className="relative col-span-4 ">
          <div className="absolute w-full border-2 border-black bg-red-100 pb-[100%]" />
        </div>
        <div className="col-span-4 m-2 flex w-full flex-col">
          <div className="flex w-full justify-center pb-3">
            <div>
              <FollowButton isFollow={true} />
            </div>
          </div>
          <div className="pb-2" />
          <div>9999ä¸‡ Follows</div>
          <div>9999 Follers</div>
        </div>
      </div>
    </div>
  );
};
