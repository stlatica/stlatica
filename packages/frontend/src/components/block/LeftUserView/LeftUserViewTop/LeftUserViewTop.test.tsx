import { render } from "@testing-library/react";
import React from "react";
import { describe, expect, test } from "vitest";

import { Story1 } from "./LeftUserViewTop.stories";


describe("snapshot", () => {
  test("snapshot", () => {
    const { asFragment } = render(<Story1 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
