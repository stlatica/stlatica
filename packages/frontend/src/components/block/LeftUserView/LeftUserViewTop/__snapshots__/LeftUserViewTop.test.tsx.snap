// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <div
      class="grid h-[170px] grid-cols-10 "
    >
      <div
        class="col-span-2 "
      >
        <div
          class="h-full"
        >
          <a
            href="/"
          >
            <div
              class="flex justify-center"
            >
              <button
                class="border-2 border-black"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="3em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="3em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </button>
            </div>
          </a>
        </div>
      </div>
      <div
        class="relative col-span-4 "
      >
        <div
          class="absolute w-full border-2 border-black bg-red-100 pb-[100%]"
        />
      </div>
      <div
        class="col-span-4 m-2 flex w-full flex-col"
      >
        <div
          class="flex w-full justify-center pb-3"
        >
          <div>
            <button
              class="text-gray-900 bg-white border border-gray-300 focus:outline-none hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:hover:bg-gray-700 dark:hover:border-gray-600 w-[8em]"
              type="button"
            >
              フォロー中
            </button>
          </div>
        </div>
        <div
          class="pb-2"
        />
        <div>
          9999万 Follows
        </div>
        <div>
          9999 Follers
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
